homeassistant/components/airzone/climate.py

```diff
@@ -105,8 +115,42 @@ def __init__(
         self._attr_hvac_modes = [
             HVAC_MODE_LIB_TO_HASS[mode] for mode in self.get_airzone_value(AZD_MODES)
         ]
+        if (
+            self.get_airzone_value(AZD_SPEED) is not None
+            and self.get_airzone_value(AZD_SPEEDS) is not None
+        ):
+            self._set_fan_speeds()
+
         self._async_update_attrs()
 
+    def _set_fan_speeds(self) -> None:
+        self._attr_supported_features |= ClimateEntityFeature.FAN_MODE
+
+        speeds = self.get_airzone_value(AZD_SPEEDS)
+        max_speed = max(speeds)
+        if max_speed == 2:
+            self._speeds = {
+                0: FAN_AUTO,
+                1: FAN_LOW,
+                2: FAN_HIGH,
```

Comment:
You could make the `dict` a constant and save a bit of branching

`
BASE_FAN_SPEEDS = {
  0: FAN_AUTO,
  1: FAN_LOW
}

FAN_SPEED_MAPS = {
   2: BASE_FAN_SPEEDS | {2: FAN_HIGH},
   3: BASE_FAN_SPEEDS | {2: FAN_MEDIUM, 3: FAN_HIGH}
}

```

Than you can do 

```python
if speeds := FAN_SPEED_MAPS.get(max_speed):
   self._speeds = speeds
else:
   for speed ....
```

Reply:
Nice one!
I will change it that way :)

Reply:
Done in ef47a1af222b45c548a573555a4aaaea09c336a1